File: tauri/apps/desktop/src/ipc/templates.ts
/**
 * Typed IPC wrappers for template operations.
 */

import type { Template, WriteReport } from "@springtale/types";
import { invoke } from "@tauri-apps/api/core";

export async function listTemplates(): Promise<Template[]> {
  return invoke<Template[]>("list_templates");
}

export async function writeTemplate(name: string): Promise<WriteReport> {
  return invoke<WriteReport>("write_template", { name });
}

File: tauri/apps/desktop/src-tauri/src/commands/templates.rs
use springtale_runtime::operations::templates::{self, Template, WriteReport};

/// List all starter project templates.
#[tauri::command]
#[specta::specta]
pub async fn list_templates() -> Result<Vec<Template>, String> {
    Ok(templates::list().to_vec())
}

/// Write a template's files to a daemon-chosen directory.
#[tauri::command]
#[specta::specta]
pub async fn write_template(name: String) -> Result<WriteReport, String> {
    templates::write_to(&name).map_err(|e| e.to_string())
}
